src/Tribe/Views/V2/Utils/Separators.php

```diff
@@ -77,4 +77,70 @@ public static function should_have_time( $events, $event ) {
 
 		return isset( $start_hours_ids[ $event_id ] );
 	}
+
+	/**
+	 * Determines if a given event from a list of events should have a type separator
+	 * for the day view template structure.
+	 *
+	 * @since  TBD
+	 *
+	 * @param  array       $events WP_Post or numeric ID for events.
+	 * @param  WP_Post|int $event  Event we want to determine.
+	 *
+	 * @return boolean
+	 */
+	public static function should_have_type( array $events, \WP_Post $event ) {
+		if ( ! is_array( $events ) ) {
+			return false;
+		}
+
+		$event_id = is_numeric( $event ) ? $event : $event->ID;
+
+		if ( empty( $event->timeslot ) ) {
+			return false;
+		}
+
+		$ids = array_map(
+			static function( $event ) {
+				return absint( is_numeric( $event ) ? $event : $event->ID );
+			},
+			$events
+		);
+
+
+		$index = array_search( $event_id, $ids );
+
+		// Return false if it wasn't found.
+		if ( false === $index ) {
+			return $index;
+		}
+
+		$is_first        = 0 === $index;
+		$is_new_timeslot = ! $is_first && $events[ $index ]->timeslot !== $events[ $index - 1 ]->timeslot;
+
+		// Should have type separator if it's the first element or if it's a new timeslot.
+		$should_have = $is_first || $is_new_timeslot;
+
+		return $should_have;
+	}
+
+	/**
+	 * Returns the text for a type separator.
+	 *
+	 * @since  TBD
+	 *
+	 * @param  WP_Post|int $event  Event we want to determine.
+	 *
+	 * @return string
+	 */
+	public static function get_type_text( \WP_Post $event ) {
```

Comment:
Lets move this into the template, no need to have a second static method on this class here.